dbt/adapters/athena/impl.py

```diff
@@ -45,26 +45,59 @@ def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
     @available
-    def s3_uuid_table_location(self):
+    def s3_table_prefix(self) -> str:
+        """
+        Returns the root location for storing tables in S3.
+        This is `s3_data_dir`, if set, and `s3_staging_dir/tables/` if not.
+        We generate a value here even if `s3_data_dir` is not set,
+        since creating a seed table requires a non-default location.
+        """
         conn = self.connections.get_thread_connection()
-        client = conn.handle
-        return f"{client.s3_staging_dir}tables/{str(uuid4())}/"
+        creds = conn.credentials
+        if creds.s3_data_dir is not None:
```

Comment:
It would have benefits if we also support setting `s3_data_dir` from the `dbt_project.yml` and not only take it from the athena connection credentials?



Example from my own experience:
- I usually make a `models/silver` and `models/gold` folder to organize data in bronze/silver/gold categories. 
- I use dbt's [custom schema feature](https://docs.getdbt.com/docs/build/custom-schemas) to suffix the database name with `_silver` and `_gold`  → the silver and gold models get deployed to their own **separate** database. 
- What would really be awesome if the silver model data can go into the silver bucket, and gold model data go into the gold bucket... 

One way to solve this would be to do:

```yaml
models:
  jesse_data:
    +materialized: table
    +s3_data_naming: schema_table
    silver:
      +s3_data_dir: s3://test-silver-jesse-test/dbt/
      +schema: silver
    gold:
      +s3_data_dir: s3://test-gold-jesse-test/dbt/
      +schema: gold
```

Or do you know a workaround to make the data dir have a dependency on the kind of model?

Reply:
This is a really good suggestion. I think that the s3_data_naming is supported by default as dbt overwrite the default confing. Regarding data_dir we could consider to also pick it from config or fall back to the profile value.
Let me play with that and come back to you.

I want also to add other 2 table scheme:
- table: only table name in the main data_dir patg
- table_unique: same as above but uuid

Reply:
I confirmed the above this already works:
```
models:
  lakehouse:
    silver:
      materialized: table
      s3_data_naming: schema_table
    iceberg:
      materialized: table
      s3_data_naming: uuid
```
Doing so I'm able to have specific behavior for group of models. I just need to add data-dir to the equation, picking from config.

Reply:
@jessedobbelaere you might want to try out again.

I tested with this in my dbt_project.yml
```
models:
  lakehouse:
    silver:
      materialized: table
      s3_data_dir: s3://my_bucket_silver/
    iceberg:
      materialized: table
      s3_data_dir: s3://my_bucket_dev/
      s3_data_naming: schema_table
```

Doing so all models under silver went to the right location, plus all models iceberg were saved somewhere else. 
Said so, each mode can be explicitly overwritten with model configuration. I guess that this solve what you would like to have.

I have other s3_data_naming convention that I would like to discuss with you btw.

Reply:
I gave it a try in one of the projects and it worked really well 👏 Data is nicely stored in either my silver or gold S3 bucket. Metadata and query results go into the staging bucket. Perfect! 🚀